### PR TITLE
Fix/did tests

### PR DESCRIPTION
**Pull request by joshLong145**
## Changes

* tweaks to validation logic to allow for fragments in did urls,
* update to validator controller tests
* removal of fragments from did urls preventing parser to resolve due to invalid characters error
* removal of object tests as they depend on network IO

## Checklist

* [x] Unit tests
* [ ] Documentation

## References *(optional)*

Fixes
Connects

 Mention [stepsize] in a comment if you'd like to report some technical debt. See examples here.

### CODE REVIEW

File: pkg/did/did.go [hunk 1]
```diff
@@ -68,7 +68,7 @@ func (d *DID) UnmarshalJSON(bytes []byte) error {
 // Address returns the Sonr Account address from the given DID
 func (d *DID) Address() string {
 	for _, str := range d.IDStrings {
-		if strings.EqualFold(str, "did") || strings.EqualFold(str, "snr") {
+		if strings.EqualFold(str, "did") || strings.EqualFold(str, "sonr") {
```
Comment (mcjcloud) [inline pkg/did/did.go]: Why is this now "sonr"?

File: pkg/did/jwe_test.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+		buf []byte
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    string
+		wantErr bool
+	}{
+		// TODO: Add test cases.
```
Comment (mcjcloud) [inline pkg/did/jwe_test.go]: No test cases

File: pkg/did/jwe_test.go [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/sonr-io/sonr/pkg/did/ssi"
+)
+
+func TestDocument_CreateJWS(t *testing.T) {
+	type fields struct {
```
Comment (mcjcloud) [inline pkg/did/jwe_test.go]: Why recreate the `Document` struct here?

File: pkg/did/jwe_test.go [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+		buf []byte
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    string
+		wantErr bool
+	}{
+		// TODO: Add test cases.
```
Comment (mcjcloud) [inline pkg/did/jwe_test.go]: Same as above ^

File: pkg/did/jwe_test.go [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+				AssertionMethod:      tt.fields.AssertionMethod,
+				KeyAgreement:         tt.fields.KeyAgreement,
+				CapabilityInvocation: tt.fields.CapabilityInvocation,
+				CapabilityDelegation: tt.fields.CapabilityDelegation,
+				Service:              tt.fields.Service,
+				AlsoKnownAs:          tt.fields.AlsoKnownAs,
+			}
+			got, err := d.CreateJWS(tt.args.buf)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Document.CreateJWS() error = %v, wantErr %v", err, tt.wantErr)
```
Comment (mcjcloud) [inline pkg/did/jwe_test.go]: You don't actually compare the error you want with the one you got

File: x/registry/types/did.go [hunk 6]
```diff
@@ -71,7 +71,7 @@ func GenerateApplicationDid(accountAddr, appToRegister string, cred *Credential)
 // GenerateNameDid generates a new did document
 func GenerateNameDid(accountAddr, nameToRegister string, cred *Credential) (*did.Document, error) {
 	// Generate a new DID String
-	baseDid, err := did.ParseDID(fmt.Sprintf("did:snr:%s", strings.TrimPrefix(accountAddr, "snr")))
+	baseDid, err := did.ParseDID(fmt.Sprintf("did:sonr:%s", strings.TrimPrefix(accountAddr, "sonr")))
```
Comment (mcjcloud) [inline x/registry/types/did.go]: Again, is `sonr` correct?

**Review by mcjcloud — CHANGES_REQUESTED** [review]

File: pkg/did/did.go (hunk 1 above)
Comment (joshLong145) [reply pkg/did/did.go]: this was a mistake, ill make this change. good catch

File: pkg/did/jwe_test.go (hunk 5 above)
Comment (joshLong145) [reply pkg/did/jwe_test.go]: good point

File: pkg/did/jwe_test.go (hunk 4 above)
Comment (joshLong145) [reply pkg/did/jwe_test.go]: good point

File: x/registry/types/did.go (hunk 6 above)
Comment (joshLong145) [reply x/registry/types/did.go]: snr is correct after reviewing prads PR

Comment (prdlk) [reply x/registry/types/did.go]: yes allows easiest namespace between wallet prefix and domain name

**Comment by joshLong145** [comment]
removed JWE related tests, will add back in in another PR after #295 merger

File: x/registry/types/did.go [hunk 7]
```diff
@@ -71,7 +71,7 @@ func GenerateApplicationDid(accountAddr, appToRegister string, cred *Credential)
 // GenerateNameDid generates a new did document
 func GenerateNameDid(accountAddr, nameToRegister string, cred *Credential) (*did.Document, error) {
 	// Generate a new DID String
-	baseDid, err := did.ParseDID(fmt.Sprintf("did:snr:%s", strings.TrimPrefix(accountAddr, "snr")))
+	baseDid, err := did.ParseDID(fmt.Sprintf("did:snr:%s", strings.TrimPrefix(accountAddr, "sonr")))
```
Comment (prdlk) [inline x/registry/types/did.go:74]: Need to fix this to be snr

File: pkg/did/did_test.go [hunk 8]
```diff
@@ -69,14 +69,17 @@ func TestParseDID(t *testing.T) {
 	})
 	t.Run("error - DID URL", func(t *testing.T) {
 		id, err := ParseDID("did:sonr:123/path?query#fragment")
```
Comment (prdlk) [inline pkg/did/did_test.go]: test string requires snr

File: pkg/did/document_test.go [hunk 9]
```diff
@@ -34,7 +34,7 @@ func Test_Document(t *testing.T) {
 			return
 		}
 		if doc.Controller[0].String() != "did:sonr:123" {
```
Comment (prdlk) [inline pkg/did/document_test.go]: Test needs to be changed to snr

**Review by prdlk — CHANGES_REQUESTED** [review]
Test methods are still using `did:sonr` opposed to `did:snr`

**Review by prdlk — DISMISSED** [review]

**Review by mcjcloud — APPROVED** [review]

**Review by ntindle — APPROVED** [review]

**Pull request merged**